File: python/7 kyu/287.py
# https://www.codewars.com/kata/58545549b45c01ccab00058c/train/python

# Passed

def calculate_total(subtotal, tax, tip):
    total = round(subtotal * (1 + (tax + tip) / 100), 2)
    return total

output = calculate_total(5.00, 5, 10)
print(output)

File: nodejs/7 kyu/287.js
// https://www.codewars.com/kata/65112af7056ad6004b5672f8/train/javascript

// Passed

function possiblyPerfect(key, answers) {
  let [correctAnswers, incorrectAnswers] = [0, 0];
  let result = true;
  for (let idx = 0; idx < key.length; idx++) {
    let studentAnswer = answers[idx];
    let correctAnswer = key[idx];

    if (correctAnswer === "_") {
      continue;
    }

    if (studentAnswer === correctAnswer) {
      correctAnswers++;
    } else {
      incorrectAnswers++;
    }

    if (correctAnswers > 0 && incorrectAnswers > 0) {
      result = false;
      break;
    }
  }

  return result;
}

const output = possiblyPerfect(
  ["A", "_", "C", "_", "B"],
  ["A", "D", "C", "E", "B"],
);
console.log(output);